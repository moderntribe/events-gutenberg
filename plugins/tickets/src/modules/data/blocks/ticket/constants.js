export const TC = 'tribe_commerce';
export const EDD = 'easy_digital_downloads';
export const WOO = 'wooCommerce';

export const TC_CLASS = 'Tribe__Tickets__Commerce__PayPal__Main';
export const EDD_CLASS = 'Tribe__Tickets_Plus__Commerce__EDD__Main';
export const WOO_CLASS = 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main';

export const PROVIDER_CLASS_TO_CONSTANT_MAPPING = {
	[ TC_CLASS ]: TC,
	[ EDD_CLASS ]: EDD,
	[ WOO_CLASS ]: WOO,
};

export const PROVIDER_TYPES = [ TC, EDD, WOO ];
