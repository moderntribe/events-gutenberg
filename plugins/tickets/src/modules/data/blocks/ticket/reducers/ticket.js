/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import * as types from './../types';
import { TYPES } from '@moderntribe/tickets/blocks/ticket/edit-container/content/capacity/template';
import { moment as momentUtil } from '@moderntribe/common/utils';

const currentMoment = moment();
export const DEFAULT_STATE = {
	title: '',
	description: '',
	price: 0,
	SKU: '',
	startDate: momentUtil.toDate( currentMoment ),
	startDateMoment: currentMoment,
	endDate: momentUtil.toDate( currentMoment ),
	endDateMoment: currentMoment,
	startTime: momentUtil.toTime24Hr( currentMoment ),
	endTime: momentUtil.toTime24Hr( currentMoment ),
	dateIsPristine: true,
	capacityType: TYPES.independent,
	capacity: '',
	isEditing: false,
	ticketId: null,
	sold: 0,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_TICKET_DESCRIPTION:
			return {
				...state,
				description: action.payload.description,
			};
		case types.SET_TICKET_PRICE:
			return {
				...state,
				price: action.payload.price,
			};
		case types.SET_TICKET_SKU:
			return {
				...state,
				SKU: action.payload.SKU,
			};
		case types.SET_TICKET_START_DATE:
			return {
				...state,
				startDate: action.payload.startDate,
			};
		case types.SET_TICKET_START_TIME:
			return {
				...state,
				startTime: action.payload.startTime,
			};
		case types.SET_TICKET_END_DATE:
			return {
				...state,
				endDate: action.payload.endDate,
			};
		case types.SET_TICKET_END_TIME:
			return {
				...state,
				endTime: action.payload.endTime,
			};
		case types.SET_TICKET_CAPACITY:
			return {
				...state,
				capacity: action.payload.capacity,
			};
		case types.SET_TICKET_CAPACITY_TYPE:
			return {
				...state,
				capacityType: action.payload.capacityType,
			};
		case types.SET_TICKET_IS_EDITING:
			return {
				...state,
				isEditing: action.payload.isEditing,
			};
		case types.SET_TICKET_ID:
			return {
				...state,
				ticketId: action.payload.ticketId,
			};
		case types.SET_TICKET_DATE_PRISTINE:
			return {
				...state,
				dateIsPristine: action.payload.dateIsPristine,
			};
		case types.SET_TICKET_START_DATE_MOMENT:
			return {
				...state,
				startDateMoment: action.payload.startDateMoment,
			};
		case types.SET_TICKET_END_DATE_MOMENT:
			return {
				...state,
				endDateMoment: action.payload.endDateMoment,
			};
		default:
			return state;
	}
};
