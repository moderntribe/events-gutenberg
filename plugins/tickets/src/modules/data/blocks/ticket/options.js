/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { SHARED, INDEPENDENT, UNLIMITED, TICKET_TYPES } from './constants'

export const CAPACITY_TYPE_OPTIONS = [
	{
		label: __( 'Share capacity with other tickets', 'events-gutenberg' ),
		value: TICKET_TYPES[ SHARED ],
	}, {
		label: __( 'Set capacity for this ticket only', 'events-gutenberg' ),
		value: TICKET_TYPES[ INDEPENDENT ],
	}, {
		label: __( 'Unlimited', 'events-gutenberg' ),
		value: TICKET_TYPES[ UNLIMITED ],
	},
];
