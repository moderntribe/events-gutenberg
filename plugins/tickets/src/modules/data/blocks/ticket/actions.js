/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const setHeader = ( header ) => ( {
	type: types.SET_TICKET_HEADER,
	payload: {
		header,
	},
} );

export const setTotalSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TOTAL_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setSettingsOpen = ( isSettingsOpen ) => ( {
	type: types.SET_TICKET_SETTINGS_OPEN,
	payload: {
		isSettingsOpen,
	},
} );

export const openSettings = () => setSettingsOpen( true );
export const closeSettings = () => setSettingsOpen( false );

export const setParentBlockSelected = ( selected ) => ( {
	type: types.SET_PARENT_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setChildBlockSelected = ( selected ) => ( {
	type: types.SET_CHILD_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setActiveChildBlockId = ( activeChildBlockId ) => ( {
	type: types.SET_ACTIVE_CHILD_BLOCK_ID,
	payload: {
		activeChildBlockId,
	},
} );

export const setTempSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TMP_TICKET_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

export const setProvider = ( provider ) => ( {
	type: types.SET_PROVIDER,
	payload: {
		provider,
	},
} );

// individual ticket actions
export const registerTicketBlock = ( blockId ) => ( {
	type: types.SET_TICKET_BLOCK_ID,
	payload: {
		blockId,
	},
} );

export const removeTicketBlock = ( blockId ) => ( {
	type: types.REMOVE_TICKET_BLOCK,
	payload: {
		blockId,
	},
} );

export const setTitle = ( blockId, title ) => ( {
	type: types.SET_TICKET_TITLE,
	payload: {
		blockId,
		title,
	},
} );

export const setDescription = ( blockId, description ) => ( {
	type: types.SET_TICKET_DESCRIPTION,
	payload: {
		blockId,
		description,
	},
} );

export const setPrice = ( blockId, price ) => ( {
	type: types.SET_TICKET_PRICE,
	payload: {
		blockId,
		price,
	},
} );

export const setSKU = ( blockId, SKU ) => ( {
	type: types.SET_TICKET_SKU,
	payload: {
		blockId,
		SKU,
	},
} );

export const setStartDate = ( blockId, startDate ) => ( {
	type: types.SET_TICKET_START_DATE,
	payload: {
		blockId,
		startDate,
	},
} );

export const setStartTime = ( blockId, startTime ) => ( {
	type: types.SET_TICKET_START_TIME,
	payload: {
		blockId,
		startTime,
	},
} );

export const setEndDate = ( blockId, endDate ) => ( {
	type: types.SET_TICKET_END_DATE,
	payload: {
		blockId,
		endDate,
	},
} );

export const setEndTime = ( blockId, endTime ) => ( {
	type: types.SET_TICKET_END_TIME,
	payload: {
		blockId,
		endTime,
	},
} );

export const setCapacityType = ( blockId, capacityType ) => ( {
	type: types.SET_TICKET_CAPACITY_TYPE,
	payload: {
		blockId,
		capacityType,
	},
} );

export const setCapacity = ( blockId, capacity ) => ( {
	type: types.SET_TICKET_CAPACITY,
	payload: {
		blockId,
		capacity,
	},
} );

export const createNewTicket = ( blockId ) => ( {
	type: types.SET_CREATE_NEW_TICKET,
	payload: {
		blockId,
	},
} );

export const updateTicket = ( blockId ) => ( {
	type: types.SET_UPDATE_TICKET,
	payload: {
		blockId,
	},
} );

export const setTicketIsEditing = ( blockId, isEditing ) => ( {
	type: types.SET_TICKET_IS_EDITING,
	payload: {
		blockId,
		isEditing,
	},
} );

export const setTicketId = ( blockId, postId ) => ( {
	type: types.SET_TICKET_ID,
	payload: {
		blockId,
		postId,
	},
} );

export const setTicketDateIsPristine = ( blockId, dateIsPristine ) => ( {
	type: types.SET_TICKET_DATE_PRISTINE,
	payload: {
		blockId,
		dateIsPristine,
	},
} );

export const setParentBlockIsLoading = ( isParentBlockLoading ) => ( {
	type: types.SET_PARENT_BLOCK_LOADING,
	payload: {
		isParentBlockLoading,
	},
} );

export const setTicketStartDateMoment = ( blockId, startDateMoment ) => ( {
	type: types.SET_TICKET_START_DATE_MOMENT,
	payload: {
		blockId,
		startDateMoment,
	},
} );

export const setTicketEndDateMoment = ( blockId, endDateMoment ) => ( {
	type: types.SET_TICKET_END_DATE_MOMENT,
	payload: {
		blockId,
		endDateMoment,
	},
} );
