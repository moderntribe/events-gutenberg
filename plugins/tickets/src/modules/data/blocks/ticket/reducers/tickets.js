/**
 * External dependencies
 */
import { combineReducers } from 'redux';
import omit from 'lodash/omit';

/**
 * Internal dependencies
 */
import * as types from './../types';
import ticket from './ticket';

const ticketsById = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_BLOCK_ID:
		case types.SET_TICKET_TITLE:
		case types.SET_TICKET_DESCRIPTION:
		case types.SET_TICKET_PRICE:
		case types.SET_TICKET_SKU:
		case types.SET_TICKET_START_DATE:
		case types.SET_TICKET_START_TIME:
		case types.SET_TICKET_END_DATE:
		case types.SET_TICKET_END_TIME:
		case types.SET_TICKET_CAPACITY_TYPE:
		case types.SET_TICKET_CAPACITY:
		case types.SET_TICKET_IS_EDITING:
		case types.SET_TICKET_ID:
		case types.SET_TICKET_DATE_PRISTINE:
		case types.SET_TICKET_START_DATE_MOMENT:
		case types.SET_TICKET_END_DATE_MOMENT:
		case types.SET_TICKET_IS_LOADING:
		case types.SET_TICKET_HAS_BEEN_CREATED:
			return {
				...state,
				[ action.payload.blockId ]: ticket( state[ action.payload.blockId ], action ),
			};
		case types.REMOVE_TICKET_BLOCK:
			return omit( state, [ action.payload.blockId ] );
		default:
			return state;
	}
};

const allTickets = ( state = [], action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_BLOCK_ID:
			return [ ...state, action.payload.blockId ];
		case types.REMOVE_TICKET_BLOCK:
			return state.filter( ( id ) => action.payload.blockId !== id );
		default:
			return state;
	}
};

export default combineReducers( {
	byId: ticketsById,
	allIds: allTickets,
} );
