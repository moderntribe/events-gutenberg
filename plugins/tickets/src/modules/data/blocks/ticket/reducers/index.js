/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import ui from './ui';
import tickets from './tickets';

export default combineReducers( {
	ui,
	tickets,
} );
