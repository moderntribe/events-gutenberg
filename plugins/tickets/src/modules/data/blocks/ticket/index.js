/**
 * Internal dependencies
 */
import reducer from './reducers';

import * as constants from './constants';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;

export { constants, types, actions, selectors, sagas };
