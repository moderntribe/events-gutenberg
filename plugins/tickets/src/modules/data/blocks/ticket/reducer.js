/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const DEFAULT_STATE = {
	sharedCapacity: '',
	header: null,
	isSettingsOpen: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_HEADER:
			return {
				...state,
				header: action.payload.header,
			};
		case types.SET_TICKET_SHARED_CAPACITY:
			return {
				...state,
				sharedCapacity: action.payload.sharedCapacity,
			};
		case types.SET_TICKET_SETTINGS_OPEN:
			return {
				...state,
				isSettingsOpen: action.payload.isSettingsOpen,
			};
		default:
			return state;
	}
}
