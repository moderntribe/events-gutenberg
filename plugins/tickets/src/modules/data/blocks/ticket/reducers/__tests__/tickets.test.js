/**
 * Internal dependencies
 */
import tickets from '../tickets';
import { actions } from '@moderntribe/tickets/data/blocks/ticket';

jest.mock( 'moment', () => () => {
	const moment = require.requireActual( 'moment' );
	return moment( 'January 10, 2018 5:30 pm', 'MMMM D, Y h:mm a' );
} );

describe( 'Tickets reducer', () => {
	let state = {};

	beforeEach( () => {
		state = { allIds: [], byId: {} };
	} );

	test( 'Default reducer', () => {
		expect( tickets( undefined, {} ) ).toEqual( state );
	} );

	test( 'Add a new block inside of the reducer', () => {
		expect( tickets( state, actions.registerTicketBlock( 'modern-tribe' ) ) ).toMatchSnapshot();
	} );

	test( 'Remove an existing block from the reducer', () => {
		state = tickets( state, actions.registerTicketBlock( 'modern-tribe' ) );
		expect( tickets( state, actions.removeTicketBlock( 'modern-tribe' ) ) ).toMatchSnapshot();
	} );
} );
