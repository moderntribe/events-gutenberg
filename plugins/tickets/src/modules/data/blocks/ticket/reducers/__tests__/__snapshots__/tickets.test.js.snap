// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tickets reducer Add a new block inside of the reducer 1`] = `
Object {
  "allIds": Array [
    "modern-tribe",
  ],
  "byId": Object {
    "modern-tribe": Object {
      "SKU": "",
      "capacity": "",
      "capacityType": "independent",
      "dateIsPristine": true,
      "description": "",
      "endDate": "January 10, 2018",
      "endDateMoment": "2018-01-10T17:30:00.000Z",
      "endTime": "17:30",
      "isEditing": false,
      "price": 0,
      "sold": 0,
      "startDate": "January 10, 2018",
      "startDateMoment": "2018-01-10T17:30:00.000Z",
      "startTime": "17:30",
      "ticketId": null,
      "title": "",
    },
  },
}
`;

exports[`Tickets reducer Remove an existing block from the reducer 1`] = `
Object {
  "allIds": Array [],
  "byId": Object {},
}
`;
